### ISSUE TITLE
Add Missing Events for Balances Pallet

### ISSUE THREAD

**Issue by muharem** [paritytech/polkadot-sdk#6974]
There are cases in the Balances pallet where events are missing, which might impact systems relying on balance updates for accounting. The known cases include:
1. `fungible` trait impl for `Hold`/`Release` operations:
No events are emitted when a balance is placed on hold or released.
2. `Burn`/`Mint` Events for `Credit`/`Debt` drops
When `Credit` or `Debt` `fungible` types are dropped, the total issuance is adjusted, but no `Burn` or `Mint` events are published.

Please verify these cases and check if any other major events are missing in the Balances pallet.

**Comment by Doordashcon** [comment]
Hello @muharem

As per my investigation:

Hold & Release Events

```rust
impl<T: Config<I>, I: 'static> fungible::MutateHold<T::AccountId> for Pallet<T, I> {

    fn done_hold(_reason: &Self::Reason, _who: &AccountId, _amount: Self::Balance) {
        Self::deposit_event(Event::<T, I>::Holding { reason, who: who.clone(), amount });
    }
    fn done_release(_reason: &Self::Reason, _who: &AccountId, _amount: Self::Balance) {
        Self::deposit_event(Event::<T, I>::Released { reason, who: who.clone(), amount });
    }

}
```

**Comment by muharem** [comment]
> Hello [@muharem](https://github.com/muharem)
> 
> As per my investigation:
> 
> Hold & Release Events
> 
> impl<T: Config<I>, I: 'static> fungible::MutateHold<T::AccountId> for Pallet<T, I> {
> 	fn done_hold(reason: &Self::Reason, who: &T::AccountId, amount: Self::Balance) {
>         Self::deposit_event(Event::<T, I>::Holding { reason: *reason, who: who.clone(), amount });
>     }
>     fn done_release(reason: &Self::Reason, who: &T::AccountId, amount: Self::Balance) {
>         Self::deposit_event(Event::<T, I>::Released { reason: *reason, who: who.clone(), amount });
>     }
> }

left comment in the PR. how about burn and mint?

**Comment by Doordashcon** [comment]
>left comment in the PR. how about burn and mint?

#### Burn
```rust
fn done_settle(who: &T::AccountId, amount: Self::Balance, dust: &Self::Balance) {
 	if dust.is_zero() {
 		Self::deposit_event(Event::<T, I>::Burned { who: who.clone(), amount: amount.clone() });
 	} else {
 		Self::deposit_event(Event::<T, I>::BurnedWithDust { who: who.clone(), amount: amount.clone(), dust: *dust});
 	}
 }

```

#### Mint
```rust


```

**Comment by muharem** [comment]
@Doordashcon its not correct. settle is not burn and resolve is not mint. you can do both without burning and minting. also I think it does not really matter if you burn 10 as amount and 0.1 as dust. you just burned 10.1

**Comment by muharem** [comment]
@Doordashcon let me know if something is not clear. I think these events publishing should be integrated into the on drops impls of `HandleImbalanceDrop` trait. But please double check this.

**Comment by Doordashcon** [comment]
@muharem 

> `Burn`/`Mint` Events for `Credit`/`Debt` drops
When `Credit` or `Debt` `fungible` types are dropped, the total issuance is adjusted, but no `Burn` or `Mint` events are published.

This a bit unclear, if the Drops handle TotalIssuance adjustments, should the event emitted be `Burn` and `Mint` or `TotalIssuanceForced`.

**Comment by muharem** [comment]
> [@muharem](https://github.com/muharem)
> 
> > `Burn`/`Mint` Events for `Credit`/`Debt` drops
> > When `Credit` or `Debt` `fungible` types are dropped, the total issuance is adjusted, but no `Burn` or `Mint` events are published.
> 
> This a bit unclear, if the Drops handle TotalIssuance adjustments, should the event emitted be `Burn` and `Mint` or `TotalIssuanceForced`.

in my option `Burn` and `Mint`. it is not forced. it seem like forced event ie meant specifically for that call under the root origin

**Comment by Yung-Beef** [comment]
Any update on this?

**Comment by Doordashcon** [comment]
This week @Yung-Beef

**Comment by JayPavlina** [comment]
We are waiting on this also. Having no events on hold and release is causing confusion for our users.
